compiler/rustc_hir_analysis/src/collect/type_of.rs

```diff
@@ -125,6 +125,10 @@ fn anon_const_type_of<'tcx>(tcx: TyCtxt<'tcx>, def_id: LocalDefId) -> Ty<'tcx> {
             return ty;
         }
 
+        Node::Field(&hir::FieldDef { default: Some(_), ty, .. }) => {
+            ItemCtxt::new(tcx, def_id).lower_ty(ty)
```

Comment:
```suggestion
        Node::Field(&hir::FieldDef { default: Some(c), def_id: field_def_id, .. }) if c.hir_id == hir_id => {
            tcx.type_of(field_def_id)
```
?

Reply:
This change game me [an ICE when evaluating the "const always panics" lint](https://github.com/rust-lang/rust/pull/129514#discussion_r1825295260) when it encounters a const generic param, but have a hack to side-step that issue that we'll need to clean up one way or another.